### ISSUE TITLE
Code standards for node-iap

### ISSUE THREAD

**Issue by KLVTZ** [Wizcorp/node-iap#33]
Whatsup!

What do you think about us adopting a coding standard for this package? 

We've been using the following:

https://github.com/standard/standard

Seems to be pretty popular among maintainers. I think it may help us with our formatting.

**Comment by ronkorving** [comment]
This package is gaining a lot of traction and is getting more and more PRs, which is awesome :)

You can tell from the inclusion of JSHint in this package that it's been around for a while, and this deserves an update ;)

I'm okay with using Standard, but would want to use https://github.com/standard/eslint-config-standard to implement it. I do believe in the value of all the other options that come with ESLint. The one drawback: `git blame`. With indentation changing, all authorship moves to whoever migrates us to Standard. I'm not particularly happy about that. One alternative is that I apply my ESLint rules, which generally match the node-iap codebase.

**Comment by KLVTZ** [comment]
@ronkorving Would it help if I add the package as a dependency and then include the config file? From there, we can work on applying the defined rules and extensions through a PR? Then you use the `standard --fix` command to apply the rules? 

I agree that the authorship should stay with the owner of the repo. Just wanted to lighten your load of work if possible.

**Comment by ronkorving** [comment]
Ah, that's very kind. It's more about credit-where-credit-is-due than ownership for us though. That is all contributions, including yours, I wouldn't want all that credit to move to a single commit that changes all tabs to spaces or whatever :)

Any style inconsistencies that may currently exist in the current codebase, would have to be updated for ESLint to pass though, but hopefully that will be kept to a minimum.

How would you feel about not adopting Standard here, but rather a set of ESLint rules (which I have lying around and can easily inject) that already match the current codestyle of the project? Thereby not messing with `git blame` output basically.

**Comment by KLVTZ** [comment]
@ronkorving yeah that sounds good! I think that is the best of both worlds. 

I do have two opinions though:

Wouldn't the commit history show who contributed a feature first --like `git log <filename>`? For example, `git blame` may show the last revision to a line of code, but looking at the `git log` of a file would show who first contributed to it. Between the `History.md` file and the release notes for a build, it does show credit to the right contributor. I personally don't use `git blame` all that often so maybe I'm missing a key feature. 

Could we adopt a max limit with our width? Some of the code seems to be a bit long. Maybe I'm just biased for 80 character widths haha. But that might also affect `git blame`

In any case, I do agree that the ESLint rules would be a good step going forward!

**Comment by ronkorving** [comment]
Good point about `git log`, it's indeed more appropriate when it comes to credit. But git blame does have its purposes ("who made this line what it is?"), and becomes harder to navigate when lines change too often (or in this case, if too many lines change for stylistic purposes only).

Yeah, let's have a max width. Personally, I find 80 very tight, but let's see what we can get away with without having to change too much code in the project :)
